api/lib/src/lib.rs

```diff
@@ -515,6 +523,72 @@ pub trait BrokerApi: SignedExtrinsicApi + StorageApi + Sized + Send + Sync + 'st
 		self.simple_submission_with_dry_run(pallet_cf_swapping::Call::deregister_as_broker {})
 			.await
 	}
+
+	async fn encode_btc_smart_contract_call(
```

Comment:
Thinking, could we add a safe mode param for this? So if safe mode is enabled, this query doesn't work. It's not bullet proof since people could construct it on their own without using this - but more than likely people will be using this, and we can prevent the creation of these memo calls